examples/simple_example.rs

```diff
@@ -41,14 +41,35 @@ extern crate libc;
 use traceme::Tracer;
 use libc::getpid;
 
+use traceme::backends::DummyTracer;
+#[cfg(perf_pt)]
+use traceme::backends::PerfPTTracer;
+
+
+/// Instantiate a tracer suitable for the current platform.
+fn init_tracer() -> Box<Tracer> {
+    if cfg!(target_os = "linux") {
+        #[cfg(perf_pt)] {
+            return Box::new(PerfPTTracer::new().trace_filename("simple_example.ptt"));
+        }
+        #[cfg(not(perf_pt))] {
+            println!("Warning: perf_pt backend not compiled in. Using dummy tracer backend.");
```

Comment:
I don't see the point in the warning: printing this out at runtime isn't really going to help users.

Reply:
I printed it because, well you don't really test anything in the case that you use the dummy backend.

How about we instead do the Rust equivalent of:
```python
def test_dummy():
    # test always run
    ...

@pytest.skipif(not supports_perf_pt())
def test_pt_perf():
    # only run if we suport pt_perf
    ...
```

i.e. test each backend separately? Then I'm happy to kill the warning (and the `init` function for that matter too).

Reply:
Yep, I understand that we can't test things in such cases. I tend to agree that some sort of "don't test things that can't be tested" dummy is a sensible sounding idea.

Reply:
So, does that mean, "yes, let's test the two backends separately"?

Reply:
We could test the dummy backend *always*, and only test the PT backend if it's enabled?

Reply:
Yep, that's exactly what I had in mind. See the example listing above.

Reply:
Then we're in agreement :)